torch/_inductor/kernel/conv.py

```diff
@@ -335,10 +338,22 @@ def convolution(
 
     x.realize()
     weight.realize()
-    layout = conv_layout(x, weight, None, **kwargs)
-    req_stride_order = ir.get_stride_order(V.graph.sizevars.size_hints(layout.stride))
-    x = ir.ExternKernel.require_stride_order(x, req_stride_order)
-    weight = ir.ExternKernel.require_stride_order(weight, req_stride_order)
+
+    # ndim can be 1 for convolution in models such as demucs
+    if V.graph.layout_opt and ndim == 2:
+        log.debug("Force channels last inputs for conv")
+        x = ir.ExternKernel.require_channels_last(x)
+        # TODO maybe we can convert weights to channels last just once before
+        # running the model.
+        weight = ir.ExternKernel.require_channels_last(weight)
+        layout = conv_layout(x, weight, None, **kwargs)
+    else:
```

Comment:
Oh, I think we already did that: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/kernel/conv.py#L302-L306 .

The reason we see a 3d tensor here is not because the tensor has no batch-size dimension, but because each channel of the tensor only has 1 dimension (rather having 2 dimensions as height and width).

I haven't checked if there is any gain in this Conv1d case yet. But I'll add a TODO here.